### DISCUSSION TITLE
please i have this same problem, I can't seem to know why

### DISCUSSION THREAD

**Discussion by liptonzuma** [jaredhanson/passport#808] [Help]
** READ THIS FIRST! **

#### Are you looking for help?

Reminder: The issue tracker is not a support forum.

Issues should only be filed in this project once they are able to be reproduced
and confirmed as a flaw in the software or incorrect information in associated
documention.

If you are encountering problems integrating this module into your application,
please post a question on the [discussion forum](https://github.com/passport/discuss)
rather than filing an issue.

#### Is this a security issue?

Do not open issues that might have security implications.  Potential security
vulnerabilities should be reported privately to jaredhanson@gmail.com.  Once any
vulerabilities have been repaired, the details will be disclosed publicly in a
responsible manner.  This also allows time for coordinating with affected parties
in order to mitigate negative consequences.

If neither of the above two scenarios apply to your situation, you should open
an issue.  Delete this paragraph and the text above, and fill in the information
requested below.

### Steps to reproduce

```js
// Format code using Markdown code blocks
```

### Environment

* Operating System: 
* Node version: 
* passport version:

**Comment by liptonzuma** [comment]
const passport = require('passport')
require('./strategies/local.strategy')

module.exports = function passportConfig(app) {
    app.use(passport.initialize())
    app.use(passport.session())

    passport.serializeUser((user, done) => {
        return done(null, user)
    });

    passport.deserializeUser((user, done) => {
        return done(null, user)
    })
}

//here is local.strategy.js

const passport = require('passport')
const { Strategy } = require('passport-local')

module.exports = function localStrategy() {
    passport.use(new Strategy({
        usernameField: 'username',
        passwordField: 'password'
    }, (username, password, done) => {
        const user = { username, password }
        return done(null, user)
    }))
}





//here is the post request



authRoutes.route('/signup').post((req, res) => {
    return req.login(req.body, () => {
        res.redirect('/auth/profile')
        console.log(req.body)
    })
})

authRoutes.route('/profile').get((req, res) => {
    res.json(req.user)
    console.log(req.user)
})

**Reply by jaredhanson** [reply]
Can you provide details as to the problem you are having?  It is hard to infer what is going wrong simply from reading the pasted code.
